src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java

```diff
@@ -171,19 +175,99 @@ public ServerQuestion insert(
       @PathVariable(name = "idPoll") int idPoll,
       @RequestBody ClientQuestion question
   ) throws ResourceNotFoundException, WrongCredentialsException {
-
-    // Retrieve the associated moderator.
-    Moderator moderator = moderatorRepository.findByToken(token)
-        .filter(m -> m.getIdModerator() == idModerator)
-        .orElseThrow(WrongCredentialsException::new);
+    if (hasAccessAsModerator(idModerator, token).isEmpty()) {
+      throw new WrongCredentialsException();
+    }
 
     // Retrieve the associated poll.
     ServerPoll poll = pollRepository.findById(ServerPollIdentifier.builder()
-        .idxModerator(moderator)
+        .idxModerator(hasAccessAsModerator(idModerator, token).get())
```

Comment:
Can refactor `hasAccessAsModerator` instead of calling it two times

Reply:
What you want to do is indeed to unwrap the `Optional` and throw an exception otherwise.